backend/src/game/routes.ts

```diff
@@ -77,7 +77,9 @@ function validate(data: any): boolean {
   if (!data.simulation || !data.simulation.length) return false;
 
   try {
-    if (!validateGame(data)) return false;
+    const game = validateGame(data);
+    if (!game) return false;
+    if (game.isGameLost() || !game.isFinished()) return false;
```

Comment:
I think this one should be inside `validateGame`, what do you think?

Reply:
We use `valiadateGame` in outro (when loading game by id). I think this is fine. validateGame checks the consistency of data. Server does some extra checks whether the game is can be displayed in the leaderboard (uses legit scenario etc)